libcodechecker/analyze/plist_parser.py

```diff
@@ -3,170 +3,185 @@
 #   This file is distributed under the University of Illinois Open Source
 #   License. See LICENSE.TXT for details.
 # -------------------------------------------------------------------------
-""""""
+"""
+Parse the plist output of an analyzer and convert it to a report for
+further processing.
 
-# FIXME: This file contains some workarounds.
-# Remove them as soon as a proper clang version comes out.
-
-import plistlib
-from xml.parsers.expat import ExpatError
-
-from libcodechecker.analyze import plist_helper
-from libcodechecker.logger import LoggerFactory
-
-LOG = LoggerFactory.get_new_logger('PLIST_PARSER')
+With the newer clang releases more information is available in the plist files.
 
+* Before Clang v3.7:
```

Comment:
I think right now it is not too much of an overhead to support plists generated by older clang versions (name detection code and hash generation). The hash generation part should be kept later for backward compatibility. The name detection is just a few simple functions so I think we should keep it, at least for the next few releases. In the upcoming v6 there will be multiple changes and we can review if we should support plists generated by older analyzer versions.